src/Service/BoardService.php

```diff
@@ -38,9 +38,9 @@ public function isPositionAvailable(array $board, array $position): bool
         return true;
     }
 
-    public function addMarble($board, $position): array
+    public function addMarble(array $board, array $position, int $value): array
     {
-        $board[$position[0]][$position[1]] = 1;
+        $board[$position[0]][$position[1]] = $value;
```

Comment:
I didn't see this before, can you clone/copy the array ? Immutability is powerful tools